### ISSUE TITLE
Adding event handlers to gatsby-image <Img /> elements

### ISSUE THREAD

**Issue by will-sparger** [gatsbyjs/gatsby#4736]
## Description

Is it possible to add React event handler functionality via props to the <Img /> element? I referenced the doc on gatsby-image as well as the sharp plugin and no event handlers are discussed or listed in the props table.

Is the current default implementation to wrap the <Img /> element inside another element (i.e. div) and add the event handler to the wrapper? This worked fine for me but I wanted to clarify if it was possible to directly add event props to <Img /> or if it were coming in a future release.
 
i.e. <Img **onClick={alert("You clicked me!")}** />

### Steps to reproduce

I tried inlining an alert (as above) as well as writing a simple event handler to be called but neither works. I wrapped the <Img /> in a div and added the click handler to it and this worked as expected.

### Expected result

The React event would fire as normal when clicking on the <Img />.

### Actual result

Nada.

### Environment

Added the images and handler code inside the about.js file of the 4th tutorial project. Not attempting in actual project at the moment.

### File contents (if changed):

None

**Comment by KyleAMathews** [comment]
With https://www.npmjs.com/package/create-react-ref we could do this for people. You interested in adding a PR?

**Comment by thebarty** [comment]
Please re-open. I find this pretty important

**Comment by ukch** [comment]
Agreed. @KyleAMathews this issue is still relevant, so can the issue please be re-opened?

**Comment by ukch** [comment]
A workaround to this problem is to wrap the `Img` in a div containing the event props:
```javascript.jsx
<div onClick={alert("You clicked me!")}>
    <Img {...props} />
</div>
```
It's not very elegant however, and results in HTML structure that looks a bit like this (lots of nested `div`s!):
```html
<div>
    <div class="gatsby-image-wrapper">
        <img ... />
        <picture>
            <img ... />
        </picture>
        <noscript>...</noscript>
    </div>
</div>
```
